src/CMakeLists.txt

```diff
@@ -52,10 +52,13 @@ SET(${PROJECT_NAME}_MULTIBODY_PARSER_PYTHON_SOURCES
   multibody/parser/python.cpp
   )
 
-SET(${PROJECT_NAME}_SOURCES
-  ${${PROJECT_NAME}_MULTIBODY_SOURCES}
-  ${${PROJECT_NAME}_MULTIBODY_PARSER_PYTHON_SOURCES}
-  )
+SET(${PROJECT_NAME}_SOURCES ${${PROJECT_NAME}_MULTIBODY_SOURCES})
+
+IF(BUILD_PYTHON_INTERFACE)
+  SET(${PROJECT_NAME}_SOURCES
+    ${PROJECT_NAME}_SOURCES
```

Comment:
You missed some curly brackets. Maybe, a better way is to append elements to list.